subiquity/client/controllers/ubuntu_advantage.py

```diff
@@ -77,21 +55,19 @@ def run_answers(self) -> None:
     def check_token(self, token: str):
         """ Asynchronously check the token passed as an argument. """
         async def inner() -> None:
-            try:
-                svcs = await \
-                        self.ua_interface.get_activable_services(token=token)
-            except InvalidUATokenError:
+            answer = await self.endpoint.check_token.GET(token)
+            if answer.status == TokenStatus.INVALID_TOKEN:
                 if isinstance(self.ui.body, UbuntuAdvantageView):
                     self.ui.body.show_invalid_token()
-            except ExpiredUATokenError:
+            elif answer.status == TokenStatus.EXPIRED_TOKEN:
                 if isinstance(self.ui.body, UbuntuAdvantageView):
                     self.ui.body.show_expired_token()
-            except CheckSubscriptionError:
+            elif answer.status == TokenStatus.UNKNOWN_ERROR:
                 if isinstance(self.ui.body, UbuntuAdvantageView):
                     self.ui.body.show_unknown_error()
             else:
                 if isinstance(self.ui.body, UbuntuAdvantageView):
-                    self.ui.body.show_activable_services(svcs)
+                    self.ui.body.show_activable_services(answer.services)
 
         self._check_task = schedule_task(inner())
 
```

Comment:
again github won't let me comment in the right place, but below here is a check "self._check_task is not None" but self._check_task is never initialized to None.